### ISSUE TITLE
Templates don't read INCLUDEs from database

### ISSUE THREAD

**Issue by ehuelsmann** [ledgersmb/LedgerSMB#854]
When the templates moved to the database, files referred to by INCLUDE directives didn't.

**Comment by ehuelsmann** [comment]
It seems that the current "load from DB" paradigm needs to be reviewed in light of the context of this section of the TT2 manual: http://www.template-toolkit.org/docs/manual/Config.html#section_LOAD_TEMPLATES

**Comment by einhverfr** [comment]
Since this is ikely to require some template changes to be really supported, I recommend setting milestone here to 1.5

**Comment by ehuelsmann** [comment]
Been studying this for a while now. I don't think there's a need to change the content of the templates, which seems like a really, really fortunate situation.

The reason for me to say so: http://template-toolkit.org/docs/modules/Template/Provider.html says
```plain
This object may be sub-classed to provide more specific facilities for loading, or otherwise providing access to templates.
```

Which means that we want to provide Template Toolkit with a different Provider -- one of our own.

**Comment by ehuelsmann** [comment]
http://template-toolkit.org/docs/modules/Template/Provider.html#section_SUBCLASSING explains exactly how to do it.

**Comment by ehuelsmann** [comment]
The above merge should have closed #854
